### PR TITLE
KRACOEUS-8828:hide s2s opportunity if non-federal sponsor is selected

### PR DESCRIPTION
**Pull request by jwillia**
this pr does a couple things
1.) hides the s2s menu item if a non-federal sponsor is selected
2.) if the sponsor is changed manually or by lookup refreshes page to update menu.
3.) if a non-federal sponsor is selected and there is an opportunity on the proposal, the opportunity is removed during navigation.
4.) fixed the progressive navigation (back, save and continue) to ignore hidden/disabled navigation menu items.

### CODE REVIEW

File: coeus-impl/src/main/java/org/kuali/coeus/propdev/impl/core/ProposalDevelopmentControllerBase.java [hunk 1]
```diff
@@ -352,6 +352,9 @@ protected void performCustomSave(DocumentBase document, Class<? extends Document
      }
      
      protected ModelAndView navigate(ProposalDevelopmentDocumentForm form, BindingResult result, HttpServletRequest request, HttpServletResponse response) throws Exception {
+         if (form.getDevelopmentProposal().getS2sOpportunity() != null && !getProposalDevelopmentService().isGrantsGovEnabledForProposal(form.getDevelopmentProposal())) {
+             ((ProposalDevelopmentViewHelperServiceImpl)form.getViewHelperService()).clearOpportunity(form.getDevelopmentProposal());
+         }
          populateAdHocRecipients(form.getProposalDevelopmentDocument());
```
Comment (gathreya) [inline coeus-impl/src/main/java/org/kuali/coeus/propdev/impl/core/ProposalDevelopmentControllerBase.java]: I wonder if this needs a warning. Does the jira say anything about this?

File: coeus-impl/src/main/java/org/kuali/coeus/propdev/impl/core/ProposalDevelopmentDocumentForm.java [hunk 2]
```diff
@@ -220,10 +220,20 @@ public int findIndexOfPageId(List<Action> actions, String pageId) {
...
-                if (component instanceof ToggleMenu) {
-                    addAllActions(actionList, ((ToggleMenu) component).getMenuItems());
+                if (component instanceof ToggleMenu && component.isRender() && !component.isHidden()) {
+                    for (Component menuComponent: ((ToggleMenu) component).getMenuItems()) {
+                        if (menuComponent instanceof Action) {
+                            Action action = (Action) menuComponent;
+                            if (!action.isDisabled() && action.isRender() && !action.isHidden()) {
+                                actionList.add(action);
+                            }
+                        }
```
Comment (gathreya) [inline coeus-impl/src/main/java/org/kuali/coeus/propdev/impl/core/ProposalDevelopmentDocumentForm.java]: I'd prefer if this went into a separate helper method.

File: coeus-impl/src/main/java/org/kuali/coeus/propdev/impl/core/ProposalDevelopmentControllerBase.java (hunk 1 above)
Comment (jwillia) [inline coeus-impl/src/main/java/org/kuali/coeus/propdev/impl/core/ProposalDevelopmentControllerBase.java]: no, the jira doesn't in 5.2.1 it also just removes the opportunity.  though a warning would be nice.

Comment (gathreya) [inline coeus-impl/src/main/java/org/kuali/coeus/propdev/impl/core/ProposalDevelopmentControllerBase.java]: If this is 5.2.1 behavior, I am OK with it.

**Comment by gathreya** [comment]
waiting for review comments to be addressed.

**Comment by jwillia** [comment]
closing and reopening off new fork

**Pull request closed without merging**